frontend/src/App.tsx

```diff
@@ -12,7 +12,11 @@ export default function App() {
   const [userEmail, setUserEmail] = useState<string | null>(null);
 
   useEffect(() => {
-    getUserEmail(setUserEmail);
+    async function getAndSetUserEmail() {
+      const userEmail = await getUserEmail();
+      setUserEmail(userEmail);
+    }
+    getAndSetUserEmail();
```

Comment:
Before this change:
`getUserEmail` does both (1) fetching the user's email and (2) setting it into the app's state.

After this change:
`getUserEmail` only fetches the user's email.

I decoupled the 2nd functionality (setting state) out of this function so that we can also use this function for getting the user's email for the Cohort Samples view's `billedBy' field.